setup.py

```diff
@@ -42,12 +42,13 @@ def has_flag(compiler, flagname):
     the specified compiler.
     """
     import tempfile
-    with tempfile.NamedTemporaryFile('w', suffix='.cpp') as f:
+    with tempfile.NamedTemporaryFile('w', suffix='.cpp', delete=False) as f:
         f.write('int main (int argc, char **argv) { return 0; }')
-        try:
-            compiler.compile([f.name], extra_postargs=[flagname])
-        except setuptools.distutils.errors.CompileError:
-            return False
+        fname = f.name
+    try:
+        compiler.compile([fname], extra_postargs=[flagname])
+    except setuptools.distutils.errors.CompileError:
+        return False
```

Comment:
Ok it makes sense.

Reply:
But in this case, the file should be deleted later.

Reply:
How about something like below, so that I don't have to worry about cleaning up stuff?

```python
    import tempfile, os
    with tempfile.TemporaryDirectory() as d:
        fname = os.path.join(d, 'main.cpp')
        with open(fname, 'w') as f:
            f.write('int main (int argc, char **argv) { return 0; }')
        try:
            compiler.compile([fname], extra_postargs=[flagname])
        except setuptools.distutils.errors.CompileError:
            return False
    return True
```

Reply:
TemporaryDirectory is not in py 2.7. Reverting to deleting the file